pkg/network/setup/podnic.go

```diff
@@ -54,7 +54,47 @@ type podNIC struct {
 	domainGenerator   domainspec.LibvirtSpecGenerator
 }
 
-func newPhase1PodNIC(vmi *v1.VirtualMachineInstance, network *v1.Network, handler netdriver.NetworkHandler, cacheCreator cacheCreator, launcherPID *int) (*podNIC, error) {
+type podNicGenerator interface {
+	generate(vmi *v1.VirtualMachineInstance, network *v1.Network, handler netdriver.NetworkHandler, cacheCreator cacheCreator, launcherPID *int) (*podNIC, error)
+	relevantNetworks(vmi *v1.VirtualMachineInstance) []v1.Network
+}
+
+type hotplugPodNicGenerator struct {
+	ifacesToHotplug []string
+}
+
+func newHotplugPodNicGenerator(ifaceToHotplug string) hotplugPodNicGenerator {
+	return hotplugPodNicGenerator{ifacesToHotplug: []string{ifaceToHotplug}}
+}
+
+func (_ hotplugPodNicGenerator) generate(vmi *v1.VirtualMachineInstance, network *v1.Network, handler netdriver.NetworkHandler, cacheCreator cacheCreator, launcherPID *int) (*podNIC, error) {
```

Comment:
The `generate` method is exactly the same in both `hotplugPodNicGenerator` and `standardPodNicGenerator`. Please use a shared implementation.

Reply:
Removed this code altogether.